### DISCUSSION TITLE
Please explain how the game handles storage

### DISCUSSION THREAD

**Discussion by airi019** [ppy/osu#28980] [Q&A] [answered]
I've been playing lazer for a few weeks and the game seems to be constantly using and relying on storage. If I'm running a few apps that make heavy use of internet such as bittorrent, then the game starts to lag during gameplay. If someone starts spectating on multiplayer during gameplay it can freeze for 1 to 3 seconds.

My question is, why the hell is the game not caching everything on ram prior to gameplay?

**Comment by peppy** [comment, marked as answer]
We cache most stuff, but for things like storyboards it's required to stream in content since it wouldn't all fit in vRAM.
